frontend/src/metabase/containers/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -0,0 +1,235 @@
+
+const setup = async ({
+  collections = COLLECTIONS,
+  dashboard = DASHBOARD,
+  noRecentDashboard = false,
+  error,
+  waitForContent = true,
+}: SetupOpts = {}) => {
+  setupCollectionsEndpoints(collections);
+  mockCollectionItemsEndpoint([dashboard]);
+  mockRootCollectionItemsEndpoint(collections, [dashboard]);
+  mockCollectionByIdEndpoint({ collections: COLLECTIONS, error });
+  setupMostRecentlyViewedDashboard(noRecentDashboard ? undefined : dashboard);
+
+  renderWithProviders(
+    <Route
+      path="/"
+      component={() => (
+        <AddToDashSelectDashModal
+          card={card}
+          onChangeLocation={() => undefined}
+          onClose={() => undefined}
+          dashboards={dashboards}
+        />
+      )}
+    />,
+    {
+      withRouter: true,
+      storeInitialState: {
+        currentUser: CURRENT_USER,
+      },
+    },
+  );
+
+  if (waitForContent) {
+    await waitForElementToBeRemoved(() => screen.queryByText("Loading..."));
+  }
+};
+
+describe("AddToDashSelectDashModal", () => {
+  describe("Create new Dashboard", () => {
+    it("should open CreateDashboardModal", async () => {
+      await setup();
+
+      const createNewDashboard = screen.getByRole("heading", {
+        name: /create a new dashboard/i,
+      });
+
+      fireEvent.click(createNewDashboard);
+
+      // opened CreateDashboardModal
+      expect(
+        screen.getByRole("heading", {
+          name: /new dashboard/i,
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  describe("Add to existing Dashboard", () => {
+    it("should show loading", async () => {
+      await setup({ waitForContent: false });
+
+      expect(screen.getByText("Loading...")).toBeInTheDocument();
+    });
+
+    it("should show error", async () => {
+      const ERROR = "Server Error!";
+      await setup({ error: ERROR });
+
+      expect(screen.getByText(ERROR)).toBeInTheDocument();
+    });
+
+    describe("when user visited some dashboard in last 24hrs", () => {
+      it("should preselected last visited dashboard in the picker", async () => {
+        await setup();
+
+        const collectionToPresent = COLLECTIONS.find(
+          collection => collection.id === DASHBOARD.collection_id,
+        );
+
+        // breadcrumbs
+        expect(
+          screen.getByText(`${collectionToPresent?.name}`),
+        ).toBeInTheDocument();
+        // dashboard item
+        expect(screen.getByText(DASHBOARD.name)).toBeInTheDocument();
+      });
+    });
+
+    describe("when user didn't visit any dashboard during last 24hrs", () => {
+      it("should render root collection without preselection", async () => {
+        await setup({
+          noRecentDashboard: true,
+        });
+
+        // breadcrumbs show root collection only
+        expect(screen.getByTestId("item-picker-header")).toHaveTextContent(
+          "Collections",
+        );
+      });
+    });
+  });
+});
+
+function mockCollectionItemsEndpoint(dashboards: Dashboard[]) {
```

Comment:
If there are no mock helpers for these endpoints please move this code to `server-mocks` and name accordingly to other helpers

Reply:
I'm afraid we'll have a lot of different helpers like this in one file which will lead to mess